node/src/types/sync_leap.rs

```
@@ -137,75 +194,88 @@ impl FetcherItem for SyncLeap {
         }
-
-        // The difference between the highest header's and the trusted header's era cannot be
-        // greater than recent_era_count. We add one, as the highest block could be a non-switch
-        // block.
         if self.signed_block_headers.len() as u64
             > chainspec.core_config.recent_era_count().saturating_add(1)
         {
             return Err(SyncLeapValidationError::TooManySwitchBlocks);
         }
-
         if self.trusted_ancestor_headers.len() as u64 > chainspec.max_blocks_per_era() {
             return Err(SyncLeapValidationError::TooManyTrustedAncestors);
         }
 
-        for signed_header in &self.signed_block_headers {
-            signed_header
-                .validate()
-                .map_err(SyncLeapValidationError::BlockWithMetadata)?;
-        }
-
         let mut headers: BTreeMap<BlockHash, &BlockHeader> = self
             .headers()
             .map(|header| (header.block_hash(), header))
             .collect();
-        let mut signatures: BTreeMap<EraId, Vec<&BlockSignatures>> = BTreeMap::new();
-        for signed_header in &self.signed_block_headers {
-            signatures
-                .entry(signed_header.block_signatures.era_id)
-                .or_default()
-                .push(&signed_header.block_signatures);
-        }
 
-        let protocol_version = chainspec.protocol_version();
-        if headers
-            .values()
-            .any(|header| header.protocol_version() != protocol_version)
-        {
+        if headers.iter().any(|(_, header)| {
+            header.protocol_version() != self.trusted_block_header.protocol_version()
+        }) {
             return Err(SyncLeapValidationError::WrongProtocolVersion);
         }
 
+        let trusted_ancestor_only = self.trusted_ancestor_only;
+        let mut signatures: Option<BTreeMap<EraId, Vec<&BlockSignatures>>>;
+        if trusted_ancestor_only {
+            signatures = None;
```

I'd return an error if `signed_block_headers` isn't empty here, and leave the rest of the code unchanced.
Why distinguish between `None` and an empty `BTreeMap`?